src/Bendec.ts

```diff
@@ -46,23 +47,23 @@ const defaultConfig: Config = {
 }
 
 // To throw better errors when type hasn't been found
-const assertExists = <T>(d: T | undefined, typeName: string) => {
+function assertExists<T>(d: T | undefined, typeName: string): asserts d is T {
   if (d === undefined) {
     throw new Error(`Type not found: '${typeName}'`)
   }
 }
 
-class Bendec<T> {
+class Bendec<DefaultType = unknown> {
```

Comment:
Currently there aren't many users of the library so we don't have to care about backwards compatibility too much